test/module/irohad/ametsuchi/flat_file_test.cpp

```diff
@@ -122,3 +126,71 @@ TEST_F(BlStore_Test, BlockStoreInitializationFromNonemptyFolder){
   // check that last ids of both block storages are the same
   ASSERT_EQ(bl_store1->last_id(), bl_store2->last_id());
 }
+
+TEST_F(BlStore_Test, EmptyDumpDir) {
+  auto res = check_consistency("");
+  ASSERT_EQ(res, nonstd::nullopt);
+}
+
+TEST_F(BlStore_Test, GetNonExistingFile) {
+  auto bl_store = FlatFile::create(block_store_path);
+  Identifier id = 98759385;  // random number that will not exist
+  auto res = bl_store->get(id);
+  ASSERT_EQ(res, nonstd::nullopt);
+}
+
+TEST_F(BlStore_Test, GetDirectory) {
+  auto bl_store = FlatFile::create(block_store_path);
+  ASSERT_EQ(bl_store->directory(), block_store_path);
+}
+
+TEST_F(BlStore_Test, GetDeniedBlock) {
+  std::vector<uint8_t> block(100000, 5);
+  auto bl_store = FlatFile::create(block_store_path);
+  auto id = 1u;
+  bl_store->add(id, block);
+
+  auto filename = boost::filesystem::path{block_store_path} / id_to_name(id);
+
+  using perms = boost::filesystem::perms;
+  auto curr = boost::filesystem::status(boost::filesystem::path(filename));
+
+  boost::filesystem::permissions(filename.string().data(), perms::no_perms);
+  auto res = bl_store->get(id);
+  ASSERT_EQ(res, nonstd::nullopt);
+  boost::filesystem::permissions(filename.string().data(), curr.permissions());
```

Comment:
What's the point of that call? The file is cleaned-up after the test

Reply:
![screenshot_20180114_000649](https://user-images.githubusercontent.com/5658339/34910066-f3186570-f8be-11e7-8447-13cb5321d12f.png)
not really

Reply:
Fixed